### PR TITLE
[Jpegd]JPEG RGB444 support HW decode on Linux

### PR DESCRIPTION
**Pull request by chuanli1**
Modify JPEG RGB444 from partial acceleration to hardware
acceleration decode

Cmd:./sample_decode jpeg -i /home/gta/IMG_0425.bmp_RGB_444_100.jpg -o test.yuv -hw
Issue: MDP-66601
Test: manually

### CODE REVIEW

**Review by wangyan-intel — APPROVED** [review]

File: _studio/shared/src/libmfx_allocator_vaapi.cpp [hunk 1]
```diff
@@ -148,6 +148,11 @@ static void FillSurfaceAttrs(std::vector<VASurfaceAttrib> &attrib, unsigned int
             break;
         case MFX_FOURCC_RGBP:
             format = VA_RT_FORMAT_RGBP;
+            attrib.resize(attrib.size()+1);
+            attrib[1].type            = VASurfaceAttribUsageHint;
+            attrib[1].flags           = VA_SURFACE_ATTRIB_SETTABLE;
+            attrib[1].value.type      = VAGenericValueTypeInteger;
+            attrib[1].value.value.i   = VA_SURFACE_ATTRIB_USAGE_HINT_DECODER;
```
Comment (dmitryermilov) [inline _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: I'm not sure that this is correct. Why HINT_DECODER?

**Review by dmitryermilov — CHANGES_REQUESTED** [review]

File: _studio/shared/src/libmfx_allocator_vaapi.cpp (hunk 1 above)
Comment (wangyan-intel) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: This Hint is used in media driver for deciding the tiling of surface. @chuanli1 Please give more detail description/comment for this.

Comment (dmitryermilov) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: It's okay. But you just hardcode this info. I mean it's just an assumption this this function is allocating surface for decoder. And the assumption can be wrong.

Comment (chuanli1) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: For decode pipeline, I need to make the value of [mediaSurface->surfaceUsageHint equal to VA_SURFACE_ATTRIB_USAGE_HINT_DECODER](https://github.com/intel/media-driver/blob/master/media_driver/linux/common/ddi/media_libva.cpp#L2509), 

[code](https://github.com/intel/media-driver/blob/master/media_driver/linux/common/ddi/media_libva_util.cpp#L231)
For RGBP, the VA_SURFACE_ATTRIB_USAGE_HINT_DECODER flag enable tileformat equal to I915_TILING_Y.

**Comment by chuanli1** [comment]
Hi @dmitryermilov,
My PR can't cause buildbot/test to fail, could you help me check it? thank you.

File: _studio/shared/src/libmfx_allocator_vaapi.cpp (hunk 1 above)
Comment (chuanli1) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: memType can‘t act on this [if condition judgment](https://github.com/intel/media-driver/blob/master/media_driver/linux/common/ddi/media_libva_util.cpp#L231)

Comment (dmitryermilov) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: Please see again: you're **unconditionally** adding hardcoded VA_SURFACE_ATTRIB_USAGE_HINT_DECODER **to a common function which has nothing about decode**. Why USAGE_HINT_DECODER? Not USAGE_HINT_ENCODER? Not TRANSCODER?

**Comment by dmitryermilov** [comment]
> Hi @dmitryermilov,
> My PR can't cause buildbot/test to fail, could you help me check it? thank you.

Please reach @eponomar . He is CI owner.

File: _studio/shared/src/libmfx_allocator_vaapi.cpp (hunk 1 above)
Comment (wangyan-intel) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: Could we use one another function for decoding only?

Comment (dmitryermilov) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: Again, why can't we do something similar what we have at lines https://github.com/Intel-Media-SDK/MediaSDK/blob/7f372a73ea73ef8a9c2fb853559a2a23a95f3e48/_studio/shared/src/libmfx_allocator_vaapi.cpp#L156-L157

Comment (wangyan-intel) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: Yes. Thanks for your comments. @chuanli1 could you please change it？

Comment (chuanli1) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp:155]: okay, thanks @wangyan-intel  and @dmitryermilov , I'll change it.

**Comment by eponomar** [comment]
> > Hi @dmitryermilov,
> > My PR can't cause buildbot/test to fail, could you help me check it? thank you.
> 
> Please reach @eponomar . He is CI owner.

Hello @chuanli1, could you please rebase and push your changes again?

**Review by dmitryermilov — APPROVED** [review]

**Comment by chuanli1** [comment]
Hi @eponomar,
Thank you. I have rebased and resubmitted my PR, but buildbot/libva build for a long time.

**Comment by chuanli1** [comment]
thank you @eponomar

**Comment by chuanli1** [comment]
Hi @dmitryermilov,
Could you help me merge this PR? Thank you.

**Pull request merged**